File: src/main/java/com/nitin/saas/member/entity/Member.java
package com.nitin.saas.member.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "members", indexes = {
        @Index(name = "idx_member_business", columnList = "businessId"),
        @Index(name = "idx_member_name", columnList = "firstName, lastName"),
        @Index(name = "idx_member_phone", columnList = "phone"),
        @Index(name = "idx_member_status", columnList = "status")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long businessId;

    @Column(nullable = false, length = 100)
    private String firstName;

    @Column(nullable = false, length = 100)
    private String lastName;

    @Column(length = 20)
    private String phone;

    @Column(length = 255)
    private String email;

    private LocalDate dateOfBirth;

    @Column(length = 10)
    private String gender;

    @Column(length = 500)
    private String address;

    @Column(nullable = false, length = 20)
    @Builder.Default
    private String status = "ACTIVE";

    @Column(columnDefinition = "TEXT")
    private String notes;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    private LocalDateTime deletedAt;

    @Version
    private Long version;


    @Column(length = 255)
    private String password;  // For member authentication

    @Column(nullable = false)
    @Builder.Default
    private Boolean accountEnabled = true;  // Can member login?

    private LocalDateTime lastLoginAt;  // Track last login time

    // ADD THIS HELPER METHOD
    public void setPasswordHash(String hashedPassword) {
        this.password = hashedPassword;
    }

    public String getFullName() {
        return firstName + " " + lastName;
    }

    public void activate() {
        this.status = "ACTIVE";
    }

    public void deactivate() {
        this.status = "INACTIVE";
        this.deletedAt = LocalDateTime.now();
    }

    public boolean isDeleted() {
        return deletedAt != null;
    }
}

File: managio_frontend/lib/types/member.ts
export type MemberStatus = 'ACTIVE' | 'INACTIVE'
export type MemberGender = 'MALE' | 'FEMALE' | 'OTHER'

export interface Member {
  id: string
  businessId: string
  firstName: string
  lastName: string
  fullName: string
  phone?: string
  email?: string
  dateOfBirth?: string
  gender?: MemberGender
  address?: string
  emergencyContactName?: string
  emergencyContactPhone?: string
  notes?: string
  status: MemberStatus
  createdAt: string
  updatedAt: string
}

export interface MemberStats {
  totalSubscriptions: number
  activeSubscriptions: number
  completedSubscriptions: number
  totalAmountPaid: number
  memberSince: string
}

export interface CreateMemberRequest {
  firstName: string
  lastName: string
  phone?: string
  email?: string
  dateOfBirth?: string
  gender?: MemberGender
  address?: string
  emergencyContactName?: string
  emergencyContactPhone?: string
  notes?: string
}

export interface UpdateMemberRequest extends Partial<CreateMemberRequest> {}